test/e2e/agentregistration_test.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Agent Registration", Label("agent-registration"), Ordered, func() {
+	It(`Should have the managed cluster "cluster-e2e-test-agent" registrated, and Available.`, func() {
+		clusterName := "cluster-e2e-test-agent"
+		By("Managed Cluster Created", func() {
+			Eventually(func() error {
+				_, err := hubClusterClient.ClusterV1().ManagedClusters().Get(context.TODO(), clusterName, metav1.GetOptions{})
+				return err
+			}, 30*time.Second, 1*time.Second).Should(Succeed())
+		})
```

Comment:
check the cluster statue and the annotations?

Reply:
The cluster is actually not avaible because we didn't enable the auto-approval in ClusterManager.
The e2e needs refactoring as we discussed, let me think it through again.

Reply:
I removed "Available" from the case description for now, I plan to refactor the e2e-prow later because it sometimes fails without any useful logs... This one may take a bit longer time.